packages/react-dom-bindings/src/client/ReactFiberConfigDOM.js

```diff
@@ -1127,6 +1127,61 @@ export function clearSuspenseBoundaryFromContainer(
   retryIfBlockedOn(container);
 }
 
+function hideOrUnhideSuspenseBoundary(
+  suspenseInstance: SuspenseInstance,
+  isHidden: boolean,
+) {
+  let node: Node = suspenseInstance;
+  // Unhide all nodes within this suspense boundary.
+  let depth = 0;
+  do {
+    const nextNode = node.nextSibling;
+    if (node.nodeType === ELEMENT_NODE) {
+      const instance = ((node: any): HTMLElement & {_stashedDisplay?: string});
+      if (isHidden) {
+        instance._stashedDisplay = instance.style.display;
```

Comment:
We don't have the original props here yet since we haven't hydrated so we don't know what value to restore it to. So we stash the old value on an expando when we hide it. This is unfortunate but also it's just an edge case.